services/cd-service/pkg/service/deploy.go

```diff
@@ -87,8 +87,12 @@ func (d *DeployServiceServer) ReleaseTrain(
 	if !valid.EnvironmentName(in.Environment) {
 		return nil, status.Error(codes.InvalidArgument, "invalid environment")
 	}
+	if in.Team != "" && !valid.TeamName(in.Team) {
+		return nil, status.Error(codes.InvalidArgument, "invalid Team name")
+	}
 	err := d.Repository.Apply(ctx, &repository.ReleaseTrain{
 		Environment: in.Environment,
+		Owner:       in.Team,
```

Comment:
I named it `team` only in the `/releasetrian?team=xyz` endpoint to be clear otherwise it named Owner which is more clear for us, but if you say it also OK to be `team` everywhere then I will rename it